pkg/api/v1/atlasbackupcompliancepolicy_types.go

```diff
@@ -0,0 +1,160 @@
+package v1
+
+import (
+	"strings"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/status"
+)
+
+func init() {
+	SchemeBuilder.Register(&AtlasBackupCompliancePolicy{}, &AtlasBackupCompliancePolicyList{})
+}
+
+// AtlasBackupCompliancePolicy defines the desired state of a compliance policy in Atlas.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories=atlas,shortName=abcp
+
+type AtlasBackupCompliancePolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AtlasBackupCompliancePolicySpec     `json:"spec,omitempty"`
+	Status status.BackupCompliancePolicyStatus `json:"status,omitempty"`
+}
+
+type AtlasBackupCompliancePolicySpec struct {
+	// Email address of the user who authorized to update the Backup Compliance Policy settings.
+	AuthorizedEmail string `json:"authorizedEmail"`
+	// First name of the user who authorized to updated the Backup Compliance Policy settings.
+	AuthorizedUserFirstName string `json:"authorizedUserFirstName"`
```

Comment:
why aren't we adding `omitempty` here and everywhere else?

Reply:
Added/updated as per the required fields listed in [the Atlas API spec](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Cloud-Backups/operation/updateDataProtectionSettings)